### PR TITLE
Az

### PR DESCRIPTION
**Pull request by andrzakharevich**

### CODE REVIEW

**Review by bettykatya — CHANGES_REQUESTED** [review]

File: azakharevich/octonion/AQA_lessons/Calc.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+def counter (x):
+    if x%2:
+        print('Введите номер ячейки для Х')
+        tmp = 'x'
+    else:
+        print('Введите номер ячейки для O')
+        tmp = 'o'
+    inp(x, tmp)
+
+def checker (x):
```
[the remark this answers is not in the document]
Comment (andrzakharevich) [reply azakharevich/octonion/AQA_lessons/Calc.py:47]: We should discuss this point

File: azakharevich/octonion/AQA_lessons/Calc.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    print(showtab())
+    x += 1
```
[the remark this answers is not in the document]
Comment (andrzakharevich) [reply azakharevich/octonion/AQA_lessons/Calc.py:49]: It's player count. X or O have to add new point

File: azakharevich/octonion/AQA_lessons/Calc.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    var = 0
+    for var in range(5):
```
[the remark this answers is not in the document]
Comment (andrzakharevich) [reply azakharevich/octonion/AQA_lessons/Calc.py:51]: Should discuss

File: azakharevich/octonion/AQA_lessons/Calc.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                    countX += 1
+                elif matrix[i][len(matrix) - i - 1] == 'o':
+                    countO += 1
+                result(countX, countO)
+        '''Строки'''
+        if var == 2:
+            for i in range(len(matrix)):
+                countX = 0
+                countO = 0
+                for j in range(len(matrix)):
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Calc.py:70]: len(matrix) is used a lot in code, why not create a variable with len(matrix) value?

**Review by bettykatya — CHANGES_REQUESTED** [review]

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+    def fullEquipPrice(self):
+        return self.itemPrice
+
+
+    def sortEquip(self):
+        print('\n\rSorting has been started...')
+
+        self.sortedArray = []
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:20]: doesn't need to be 'self'

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                self.currEquip = self.equipArray[i]
+                if self.currEquip.weight < self.minEquip.weight:
+                    self.minEquip = self.currEquip
+                    self.minIndex = i
+            self.sortedArray.append(self.minEquip)
+            self.equipArray.pop(self.minIndex)
+        self.equipArray = self.sortedArray
+
+        print(' After:')
+        for i in range(len(self.equipArray)): #Помню, что если использовали 2 раза, то можно вынести в отдельный метод, но копипаст тут быстрее ;)
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:39]: nope, it is not faster)

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        print('\n\rSorting has been started...')
+
+        self.sortedArray = []
+
+        print(' Before:')
+        for i in range(len(self.equipArray)):
+            print('-', self.equipArray[i].weight)
+
+        while len(self.equipArray) > 0:
+            self.minEquip = self.equipArray[0]
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:27]: doesn't need to be 'self'

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            self.minIndex = 0
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:28]: doesn't need to be 'self'

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+            for i in range(len(self.equipArray)):
+                self.currEquip = self.equipArray[i]
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:30]: doesn't need to be 'self'

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+        print(' After:')
+        for i in range(len(self.equipArray)): #Помню, что если использовали 2 раза, то можно вынести в отдельный метод, но копипаст тут быстрее ;)
+            print('-', self.equipArray[i].weight)
+
+
+    def findEquip(self, min, max):
+        self.min = min
+        self.max = max
+        self.searchResult = []
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:46]: doesn't need to be 'self'

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+            for i in range(len(self.equipArray)):
+                self.currEquip = self.equipArray[i]
+                if self.currEquip.weight < self.minEquip.weight:
+                    self.minEquip = self.currEquip
+                    self.minIndex = i
+            self.sortedArray.append(self.minEquip)
+            self.equipArray.pop(self.minIndex)
+        self.equipArray = self.sortedArray
+
+        print(' After:')
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:38]: i would move both prints (before and after) to separate method that is called from main

**Review by bettykatya — CHANGES_REQUESTED** [review]

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+    def findEquip(self, min, max):
+        self.min = min
+        self.max = max
+        self.searchResult = []
+        for i in range(len(self.equipArray)):
+            if min < self.equipArray[i].price < max:
+                self.searchResult.append(self.equipArray[i])
+
+        print('\n\rSearch equip from', self.min,'to', self.max, '$')
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:51]: it would be better to return list and print it outside

**Review by bettykatya — CHANGES_REQUESTED** [review]

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+
+    def equipWeightList(self):
+        for i in range(len(self.equipArray)):
+            print('-', self.equipArray[i].weight)
+
+
+    def findEquip(self, min, max):
+        self.min = min
+        self.max = max
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:40]: please check all classes for unneeded use of 'self'

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+                currEquip = self.equipArray[i]
+                if currEquip.weight < minEquip.weight:
+                    minEquip = currEquip
+                    minIndex = i
+            sortedArray.append(minEquip)
+            self.equipArray.pop(minIndex)
+        self.equipArray = sortedArray
+
+
+    def equipWeightList(self):
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:33]: rename

File: azakharevich/octonion/AQA_lessons/Class/person.py [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+        self.itemPrice = 0
+        self.equipArray = []
+
+
+    def addEquip(self, object):
+        self.itemPrice += object.price
+        self.equipArray.append(object)
+
+
+    def fullEquipPrice(self):
```
Comment (bettykatya) [inline azakharevich/octonion/AQA_lessons/Class/person.py:13]: rename

**Review by bettykatya — CHANGES_REQUESTED** [review]

**Review by bettykatya — CHANGES_REQUESTED** [review]

**Review by bettykatya — APPROVED** [review]

**Pull request merged**